tests/telemetry/test_telemetry.py

```diff
@@ -182,8 +186,46 @@ def test_virtualdb_table_streaming(duthosts, enum_rand_one_per_hwsku_hostname, p
                  "Timestamp markers for each update message in:\n{0}".format(result))
 
 
-def invoke_py_cli_from_ptf(ptfhost, cmd):
-    ptfhost.shell(cmd)
+def invoke_py_cli_from_ptf(ptfhost, cmd, results=[""], match_no=0, find_data=""):
+    gnmi_output = ptfhost.shell(cmd)['stdout']
+    gnmi_str = str(gnmi_output)
+    if find_data != "":
+        result = fetch_json_ptf_output(gnmi_str, match_no)
+        match = re.findall(find_data, result)
+        assert len(match) > 0
+        results[0] = match[0]
+
+
+def test_on_change_updates(duthosts, enum_rand_one_per_hwsku_hostname, ptfhost, localhost, gnxi_path):
+    logger.info("Testing on change update notifications")
+
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    skip_201911_and_older(duthost)
+    cmd = generate_client_cli(duthost=duthost, gnxi_path=gnxi_path, method=METHOD_SUBSCRIBE,
+                              submode=SUBMODE_ONCHANGE, update_count=2, xpath="NEIGH_STATE_TABLE",
+                              target="STATE_DB")
+    results = [""]
+
+    bgp_nbrs = list(duthost.get_bgp_neighbors().keys())
+    bgp_neighbor = random.choice(bgp_nbrs)
+    bgp_info = duthost.get_bgp_neighbor_info(bgp_neighbor)
+    original_state = bgp_info["bgpState"]
+    new_state = "Established" if original_state.lower() == "active" else "Active"
+
+    client_thread = threading.Thread(target=invoke_py_cli_from_ptf, args=(ptfhost, cmd, results, 1, bgp_neighbor,))
+    client_thread.start()
+
+    wait_until(5, 1, 0, check_gnmi_cli_running, ptfhost)
+    duthost.shell("sonic-db-cli STATE_DB HSET \"NEIGH_STATE_TABLE|{}\" \"state\" {}".format(bgp_neighbor,
+                                                                                            new_state))
+
+    client_thread.join(30)
```

Comment:
>30 [](http://example.com/codeflow?start=23&length=2)

If timeout happens, is it a failure?

Reply:
Yes, it will fail. Gnmi_cli process will exit and assert will throw error as output was not received.